### ISSUE TITLE
[Bug]: Unable to tab to the End date in DatePicker component on selecting a start date

### ISSUE THREAD

**Issue by kvaidyanathan-ns1** [carbon-design-system/carbon#18397]
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.72.0

### Description

During an accessibility check, I observed that while I can select a start date in the DatePicker component, I am unable to tab to the end date. Instead, the focus is redirected to the top of the page. Please refer to the attached video for a demonstration.

Expected Behavior:
After selecting the start date, I should be able to tab through seamlessly and select the end date.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-datepicker--overview#range-datepicker

[Stackblitz Example
](https://stackblitz.com/edit/2qh31d8a?file=src%2FApp.jsx)
### Steps to reproduce

Navigate to the start date field.

- Press the down arrow key to open the calendar.
- Select a date by pressing the Enter key.
- Attempt to tab to the end date field.

Expected: The focus should move to the end date field when tabbing.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by YaswanthManoharan** [comment]
@kvaidyanathan-ns1 I tried to reproduce the same. But for me when clicking on tab after selecting a date in start date, Its moving to end date field as expected. Can you please check again.
https://react.carbondesignsystem.com/?path=/docs/components-datepicker--overview
https://stackblitz.com/edit/2qh31d8a?file=src%2FApp.jsx

In both the cds docs and the stackblitz, its as expected.

**Comment by kvaidyanathan-ns1** [comment]
@YaswanthManoharan  Could you try reproducing in the below StackBlitz link? I confirmed the same behavior on a colleague's system and was able to reproduce it:
https://stackblitz.com/edit/2qh31d8a-qmzbys5h?file=src%2FApp.jsx

https://github.com/user-attachments/assets/bc6714d3-bbcd-45f1-83d3-317622518a17

**Comment by annawen1** [comment]
Hi @kvaidyanathan-ns1 I'm not able to reproduce the issue. Steps I'm doing:

1. Tab into the start date input
2. Down arrow key to shift focus to calendar
3. arrow keys to pick desired date
4. Enter to select date
5. Escape key to return focus back to start date input
6. Tab to shift focus to end date input

https://github.com/user-attachments/assets/cfe7cfd3-f00e-4c85-8355-ccc98dfbb8dc

**Comment by KJaspers** [comment]
We're also having issues with the keyboard operation of the date picker. When trying out the Stackblitz example from the problem description, I was able to press the Tab key after changing the date to move to the next element (see Video 1). BUT when using `allowInput={false}` (this is how our production code is implemented), I have trouble leaving the start date input field using the keyboard (trouble = it didn't work with Tab, Shift+Tab, or Esc) (Video 2).

Video 1:
https://github.com/user-attachments/assets/1a7e96c3-c012-48cc-ae7f-4b5282dfba3e

Video 2:
https://github.com/user-attachments/assets/59e64194-a256-4de0-ac90-0816ef36139e

**Comment by KJaspers** [comment]
Hi everyone, any updates on this? This is what is preventing us from being fully accessible...

**Comment by Geendersom** [comment]
Hi! I'd like to work on this issue as my first contribution to Carbon.

I've reproduced the issue and would like to investigate the focus management after selecting the start date in the DatePicker range component.

If there are no objections, I'd be happy to work on a fix and add the appropriate tests.
